Homework_11/src/Circus.java

```diff
@@ -0,0 +1,61 @@
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Circus<T extends Performer> {
+    private final List<T> performers;
+
+    public Circus() {
+        this.performers = new ArrayList<>();
+    }
+
+    public void addPerformer(T performer) {
+        performers.add(performer);
+    }
+
+    public void removePerformer(T performer) {
+        performers.remove(performer);
+    }
+
+    public void removeAllPerformers() {
+        performers.clear();
+    }
+
+    public List<T> listPerformers() {
+        return performers;
+    }
+
+    public void saveToFile(String filename) {
+        try (BufferedWriter bw = new BufferedWriter(new FileWriter(filename))) {
+            for (T performer : performers) {
+                if (performer instanceof CircusPerformer) {
+                    CircusPerformer currentPerformer = (CircusPerformer) performer;
+                    bw.write(currentPerformer.getName() + "," + currentPerformer.getPerformanceType() + "," + currentPerformer.getExperience() + "\n");
```

Comment:
The approach you've used is good enough (and most of students did it like that), although as we're basically doing the serialization of an object - you can try the direct serialization/deserialization if it has caught your interest :)

https://www.geeksforgeeks.org/serializable-interface-in-java/

The only thing is that CircusPerformer is not a classic DTO/POJO, so you this might require additional tricks to make it work properly.